crates/oxc_linter/src/rules/unicorn/prefer_optional_catch_binding.rs

```diff
@@ -51,7 +51,29 @@ impl Rule for PreferOptionalCatchBinding {
         if references_count != 0 {
             return;
         }
-        ctx.diagnostic(prefer_optional_catch_binding_diagnostic(catch_param.pattern.span()));
+        let Some(parent_node) = ctx.nodes().parent_node(node.id()) else {
+            return;
+        };
+        let AstKind::CatchClause(catch_clause) = parent_node.kind() else {
+            return;
+        };
+        ctx.diagnostic_with_fix(
+            prefer_optional_catch_binding_diagnostic(catch_param.pattern.span()),
+            |fixer| {
+                let mut start = catch_clause.span().start + 5;
+                let total_param = Span::new(start, catch_param.span().start);
+                let value = ctx.source_range(total_param);
+                for char in value.chars() {
```

Comment:
Spans cover bytes, not chars. Also, since this breaks on non-whitespace characters, and we always know that a space will be a single byte, it is safe and faster to iterate over bytes here

Reply:
Consider something like
```rs
let span = Span::new(start, catch_param.span().start);
let amount_to_expand_end_by = value.as_bytes().iter().take_while(char::is_whitespace).count();
let span = span.expand_right(amount_to_expand_end_by);
```

This code may not match your desired logic exactly, but the technique should be the same.

Reply:
> Consider something like
> 
> ```rust
> let span = Span::new(start, catch_param.span().start);
> let amount_to_expand_end_by = value.as_bytes().iter().take_while(char::is_whitespace).count();
> let span = span.expand_right(amount_to_expand_end_by);
> ```
> 
> This code may not match your desired logic exactly, but the technique should be the same.

I made some changes. I need to handle the change from usize to u32 here. My handling method may not be simple enough, welcome to commit.